migrator/migrations/m_196_to_m_197_set_poduid_where_null/store/pod/store.go

```diff
@@ -0,0 +1,287 @@
+
+	query = "delete from pods_live_instances where pods_Id = $1 AND idx >= $2"
+	batch.Queue(query, pgutils.NilOrUUID(obj.GetId()), len(obj.GetLiveInstances()))
+	return nil
+}
+
+func insertIntoPodsLiveInstances(batch *pgx.Batch, obj *storage.ContainerInstance, podID string, idx int) error {
+
+	values := []interface{}{
+		// parent primary keys start
+		pgutils.NilOrUUID(podID),
+		idx,
+		obj.GetImageDigest(),
+	}
+
+	finalStr := "INSERT INTO pods_live_instances (pods_Id, idx, ImageDigest) VALUES($1, $2, $3) ON CONFLICT(pods_Id, idx) DO UPDATE SET pods_Id = EXCLUDED.pods_Id, idx = EXCLUDED.idx, ImageDigest = EXCLUDED.ImageDigest"
+	batch.Queue(finalStr, values...)
+
+	return nil
+}
+
+func copyFromPods(ctx context.Context, s pgSearch.Deleter, tx *postgres.Tx, objs ...*storage.Pod) error {
+	inputRows := make([][]interface{}, 0, batchSize)
+
+	// This is a copy so first we must delete the rows and re-add them
+	// Which is essentially the desired behaviour of an upsert.
+	deletes := make([]string, 0, batchSize)
+
+	copyCols := []string{
+		"id",
+		"name",
+		"deploymentid",
+		"namespace",
+		"clusterid",
+		"serialized",
+	}
+
+	for idx, obj := range objs {
+		// Todo: ROX-9499 Figure out how to more cleanly template around this issue.
+		log.Debugf("This is here for now because there is an issue with pods_TerminatedInstances where the obj "+
+			"in the loop is not used as it only consists of the parent ID and the index.  Putting this here as a stop gap "+
+			"to simply use the object.  %s", obj)
+
+		serialized, marshalErr := obj.Marshal()
+		if marshalErr != nil {
+			return marshalErr
+		}
+
+		inputRows = append(inputRows, []interface{}{
+			pgutils.NilOrUUID(obj.GetId()),
+			obj.GetName(),
+			pgutils.NilOrUUID(obj.GetDeploymentId()),
+			obj.GetNamespace(),
+			pgutils.NilOrUUID(obj.GetClusterId()),
+			serialized,
+		})
+
+		// Add the ID to be deleted.
+		deletes = append(deletes, obj.GetId())
+
+		// if we hit our batch size we need to push the data
+		if (idx+1)%batchSize == 0 || idx == len(objs)-1 {
+			// copy does not upsert so have to delete first.  parent deletion cascades so only need to
+			// delete for the top level parent
+
+			if err := s.DeleteMany(ctx, deletes); err != nil {
+				return err
+			}
+			// clear the inserts and vals for the next batch
+			deletes = deletes[:0]
+
+			if _, err := tx.CopyFrom(ctx, pgx.Identifier{"pods"}, copyCols, pgx.CopyFromRows(inputRows)); err != nil {
+				return err
+			}
+			// clear the input rows for the next batch
+			inputRows = inputRows[:0]
+		}
+	}
+
+	for idx, obj := range objs {
+		_ = idx // idx may or may not be used depending on how nested we are, so avoid compile-time errors.
+
+		if err := copyFromPodsLiveInstances(ctx, s, tx, obj.GetId(), obj.GetLiveInstances()...); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func copyFromPodsLiveInstances(ctx context.Context, s pgSearch.Deleter, tx *postgres.Tx, podID string, objs ...*storage.ContainerInstance) error {
```

Comment:
Style check fails on this line as `s` is not used. I don't want to modify this file as it is copied and I don't understand why the original file passes style check. Perhaps the original is exempted from the style check.